src/main/java/me/imoltres/bbu/game/threads/GameThread.kt

```diff
@@ -40,6 +45,17 @@ class GameThread(val game: Game) : Thread() {
                 checkWinConditions()
 
                 when (game.gameState) {
+                    GameState.PRE_GAME -> { //This should probably be more efficient
+                        //check and see if all players are in their cages
+                        for (player in game.getAliveTeams().flatMap { it.players }) {
```

Comment:
This loop is not okay, when checking a position in a cuboid, it runs a O(n^3) loop, running this function inside of another loop makes this O(n^4) PLEASE NO

Reply:
Actually Cuboid#contains is O(1) not O(n^3) so this would make it O(n).

Reply:
misread it as contents, my bad!

Reply:
All good :P